src/lib/error/bad-data-error.ts

```diff
@@ -121,3 +122,29 @@ export const fromOpenApiValidationErrors = (
         [firstDetail, ...remainingDetails],
     );
 };
+
+export const fromJoiError = (err: ValidationError): BadDataError => {
+    const details = err.details.map((detail) => {
+        const messageEnd = detail.context?.value
+            ? `. You provided ${JSON.stringify(detail.context.value)}.`
+            : '.';
+        const description = detail.message + messageEnd;
+        return {
+            description,
+            message: description,
+        };
+    });
+
+    const [first, ...rest] = details;
```

Comment:
Why do we need this and check for `first` afterwards? 🤔 
Couldn't we check `details.length` on line 140 and pass in `details` in the second parameter in 143?

Reply:
Good question! I've tried that multiple times, but typescript doesn't accept that, I'm afraid 🤷🏼